job/server/src/main/java/alluxio/master/job/command/CommandManager.java

```diff
@@ -90,8 +88,27 @@ public synchronized void submitCancelTaskCommand(long jobId, long taskId, long w
     cancelTaskCommand.setTaskId(taskId);
     JobCommand.Builder command = JobCommand.newBuilder();
     command.setCancelTaskCommand(cancelTaskCommand);
+    submit(workerId, command);
+  }
+
+  /**
+   * Submits a set thread pool size command to specific worker.
+   *
+   * @param workerId the worker id
+   * @param taskPoolSize the task pool size
+   */
+  public synchronized void submitSetTaskPoolSizeCommand(long workerId, int taskPoolSize) {
+    SetTaskPoolSizeCommand.Builder setTaskPoolSizeCommand = SetTaskPoolSizeCommand.newBuilder();
+    setTaskPoolSizeCommand.setTaskPoolSize(taskPoolSize);
+
+    JobCommand.Builder command = JobCommand.newBuilder();
+    command.setSetTaskPoolSizeCommand(setTaskPoolSizeCommand);
+    submit(workerId, command);
+  }
+
+  private void submit(long workerId, JobCommand.Builder command) {
     if (!mWorkerIdToPendingCommands.containsKey(workerId)) {
-      mWorkerIdToPendingCommands.put(workerId, Lists.<JobCommand>newArrayList());
+      mWorkerIdToPendingCommands.put(workerId, Lists.newArrayList());
```

Comment:
Is this already protected by a lock?

Reply:
Every other public method caller to this is synchronized so since it doesn't make a difference, let me add synchronized to here as well.